src/cdk/a11y/focus-trap/focus-trap.ts

```diff
@@ -194,6 +225,42 @@ export class FocusTrap {
         markers[markers.length - 1] : this._getLastTabbableElement(this._element);
   }
 
+  /**
+   * Focuses the provided element. If the element is not focusable, it will add a tabIndex
+   * attribute to forcefully focus it. The attribute is removed after focus is moved.
+   * @param element The element to focus.
+   */
+  forceFocus(element: HTMLElement, options?: FocusOptions) {
+    if (!this._checker.isFocusable(element)) {
+      element.tabIndex = -1;
+      // The tabindex attribute should be removed to avoid navigating to that element again
+      element.addEventListener('blur', () => element.removeAttribute('tabindex'));
```

Comment:
This handler should be added outside of the Angular zone. You can look through the codebase for `runOutsideAngular` for an example.